test/e2e.nut.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { use, expect } from 'chai';
+import * as chaiEach from 'chai-each';
+
+import { TestSession, execCmd } from '@salesforce/cli-plugins-testkit';
+
+import { PermsetAssignResult } from '../src/commands/force/user/permset/assign';
+import { AuthList } from '../src/commands/force/user/list';
+import { UserCreateOutput } from '../src/commands/force/user/create';
+
+use(chaiEach);
+const projectPath = 'testProject_AllUserCommandsWithJSON';
+let session: TestSession;
+
+describe('verifies all commands run successfully ', () => {
+  before(() => {
+    session = TestSession.create({
+      project: {
+        destinationDir: projectPath,
+        sourceDir: 'test/df17AppBuilding',
+      },
+      // create org and push source to get a permset
+      setupCommands: ['sfdx force:org:create -d 1 -s -f config/project-scratch-def.json', 'sfdx force:source:push'],
+    });
+  });
+
+  it('user display', () => {
+    const output = execCmd('force:user:display --json', { ensureExitCode: 0 });
+    expect(output.jsonOutput)
+      .to.have.property('result')
+      .with.all.keys(['username', 'accessToken', 'id', 'orgId', 'profileName', 'loginUrl', 'instanceUrl']);
+    const result = (output.jsonOutput as Record<string, unknown>).result as Record<string, string>;
```

Comment:
This casting is kind of a bummer.  I wonder if we should augment `execCmd` to return results as a certain type.  Something like: `execCmd<UserDisplayResult>(...);` So that result is typed for us.

Reply:
that's a really great idea